### ISSUE TITLE
IE redirects to logout after a few seconds

### ISSUE THREAD

**Issue by cletisgipson** [IdentityServer/IdentityServer3#1209]
IE redirects to logout after a few seconds.  This same functionality works in Chrome and Firefox.  Any ideas what might be going on?

**Comment by brockallen** [comment]
You need to provide better context.

**Comment by cletisgipson** [comment]
I setup the OIDC.js  single sign out functionality as it was setup in the samples.  

 

The single sign out feature works flawlessly in Chrome and Firefox.  I can log in to multiple apps on the same browser, when I sign out of one, they are all automatically signed out.  

 

In IE however, as soon as I sign in and redirected back to the exact same application that I have pulled up in Chrome or FF, after about 5 seconds, I get redirected back to the sign out confirmation on the IDP.

**Comment by cletisgipson** [comment]
I have tracked it down a bit more.  The session state is making it's way to the rp frame.  From there it is sent to the OP frame.  It is passing all the required checks. It is failing the check for the hash that is sent in vs the expected hash.  Each time sending back a changed response.  This is only happening in IE, both 10 and 11.  I also tried different view modes in IE, all with the same outcome.  Is there a way to troubleshoot the check_session script on the IDP side?  I want to see what values it is expecting vs what it is getting.
